backend/app/services/schwab_csv.py

```
@@ -172,14 +172,32 @@ def _map_csv_row(raw_row: dict, header_map: dict[str, str]) -> dict | None:
-    # Premium per share: abs(Amount) / (quantity * 100). Schwab's "Amount"
-    # column is signed (positive = credit, negative = debit) and already net of
-    # fees, mirroring the API's `netAmount` field.
-    premium_per_share = abs(amount) / (quantity * 100) if quantity > 0 else 0.0
+    # Premium per share: prefer the per-share ``Price`` column when present —
+    # Schwab's CSV exposes the gross premium there directly, which sidesteps
+    # the fee-arithmetic that ``Amount`` is already net of. Fall back to
+    # gross-up math (``abs(Amount) + fees``) when ``Price`` is missing or
+    # unparseable — that path applies to assignment / expired rows that have
+    # no Price column populated, where the fallback produces 0.0 from a 0.0
+    # amount + 0.0 fees anyway. See issue #184.
+    # Buys pay gross + fees; sells receive gross − fees. Fee sign in the
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)